src/webswview/lkswview/api/views.py

```diff
@@ -16,4 +25,64 @@ def get(self, id):
         return {'task': 'Say "Hello, World!"'}
 
 
+class SrcPackage(Resource):
+    def get(self, repo_name, suite_name, name):
+        with session_scope() as session:
+            rss = repo_suite_settings_for(session, repo_name, suite_name, fail_if_missing=False)
+            if not rss:
+                abort(404)
+
+            spkgs = (
+                session.query(SourcePackage)
+                .options(undefer(SourcePackage.version))
+                .filter(SourcePackage.repo_id == rss.repo_id)
+                .filter(SourcePackage.suites.any(ArchiveSuite.id == rss.suite_id))
+                .filter(SourcePackage.name == name)
+                .filter(SourcePackage.time_deleted.is_(None))
+                .order_by(SourcePackage.version.desc())
+                .all()
+            )
+            if not spkgs:
+                abort(404)
+
+            repo = rss.repo
+            suite = rss.suite
+
+            return {
+                'pkgs': [
+                    {
+                        'uuid': spkg.uuid,
+                        'source_uuid': spkg.source_uuid,
+                        'name': spkg.name,
+                        'version': spkg.version,
+                        # skipping repo because presumably it's same as rss.repo
+                        # skipping suites because there's no good reason to have them in a suite-specific query
+                        'time_added': spkg.time_added,
+                        'time_published': spkg.time_published,
+                        'architectures': spkg.architectures,
+                        'maintainer': spkg.maintainer,
+                        'original_maintainer': spkg.original_maintainer,
+                        'uploaders': spkg.uploaders,
+                        'changes_urgency': spkg.changes_urgency,
+                    }
+                    for spkg in spkgs
+                ],
+                'pkg_repo': {
+                    'name': repo.name,
+                    'origin_name': repo.origin_name,
+                },
+                'pkg_suite': {
```

Comment:
Shouldn't queries to repository and suite details be separate API endpoints? That would be more efficient (we don't sent this info over the wire all the time) and be a bit more extensible too!

Reply:
Given that there were no other API users, I was building this more for my specific use case :). The only difficulty I see is that two requests would double the roundtrip delay or complicate dispatch. But that's not a big problem.

Reply:
Suites are static, so any client can make it once and cache the result :-)
And I'd rather have the extensible API now than dealing with the consequences of an API break later...